helper/src/main/java/me/lucko/helper/utils/XMaterial.java

```diff
@@ -0,0 +1,1686 @@
+     */
+    public static boolean isNewVersion() {
+        if (isNewVersion != null) return isNewVersion;
+        return isNewVersion = isVersionOrHigher(MinecraftVersion.VERSION_1_13);
+    }
+
+    public static boolean isOneEight() {
+        return getVersion() == MinecraftVersion.VERSION_1_8;
+    }
+
+    /**
+     * Uses newly added materials to minecraft to detect the server version.
+     *
+     * @return the current server version.
+     */
+    @Nonnull
+    public static MinecraftVersion getVersion() {
+        if (version != null) return version;
+        return version = valueOfVersion(Bukkit.getVersion());
+    }
+
+    /**
+     * When using newer versions of Minecraft {@link #isNewVersion()} this helps<br>
+     * to find the old material name with its data using a cached search for optimization.
+     *
+     * @see #matchXMaterial(String, byte)
+     */
+    @Nonnull
+    private static XMaterial requestOldXMaterial(@Nonnull String name, byte data) {
+        XMaterial cached = CACHED_SEARCH.get(name + "," + data);
+
+        if (cached != null) return cached;
+        Optional<XMaterial> search = data == -1 ? Arrays.stream(XMaterial.VALUES).filter(mat -> mat.matchAnyLegacy(name)).findFirst()
+                : Arrays.stream(XMaterial.VALUES).filter(mat -> mat.matchAnyLegacy(name) && mat.data == data).findFirst();
+
+        if (search.isPresent()) {
+            XMaterial found = search.get();
+            CACHED_SEARCH.put(found.legacy[0] + "," + found.getData(), found);
+            return found;
+        }
+        return null;
+    }
+
+    /**
+     * Checks if XMaterial enum contains a material with this name.
+     *
+     * @param name name of the material
+     * @return true if XMaterial enum has this material.
+     */
+    public static boolean contains(@Nonnull String name) {
+        Validate.notEmpty(name, "Material name cannot be null or empty");
+        String formatted = format(name);
+        return Arrays.stream(XMaterial.VALUES).anyMatch(mat -> mat.name().equals(formatted));
+    }
+
+    /**
+     * Checks if the given material matches any of the legacy names.<br>
+     * <b>Not recommended to use as it may cause a little performance decrease.</b>
+     *
+     * @param name the material name.
+     * @return true if it's a legacy name.
+     * @see #matchAnyLegacy(String)
+     */
+    public static boolean containsLegacy(@Nonnull String name) {
+        Validate.notEmpty(name, "Material name cannot be null or empty");
+        String formatted = format(name);
+        return Arrays.stream(LEGACY_VALUES).anyMatch(mat -> parseLegacyVersionMaterialName(mat).equals(formatted));
+    }
+
+    /**
+     * @see #matchXMaterial(String, byte)
+     */
+    @Nullable
+    public static XMaterial matchXMaterial(@Nonnull Material material) {
+        return matchXMaterial(material.name());
+    }
+
+    /**
+     * Can also be used like: <b>MATERIAL:DATA</b>
+     * E.g. <b>RED_WOOL:14</b> or <b>WOOL:14</b>
```

Comment:
Additional paragraphs in comments should include an empty line between previous paragraphs.